src/SANDECALClustering.cpp

```diff
@@ -0,0 +1,92 @@
+#include "SANDClustering.h"
+
+int clustering(std::string const& input)
+{
+  
+  const char* finname = input.c_str();
+  TFile f(finname, "READ");
+  TTree* t = (TTree*)f.Get("tDigit");
+  int nEvents = t->GetEntries();
+  std::vector<dg_cell>* cell = new std::vector<dg_cell>;
+  std::vector<cluster> f_clust, og_clust;
+  TString output = input;
+  output.ReplaceAll(".root", ".Clusters.root");
+  TFile fout(output, "RECREATE");
+  TTree tout("tCluster", "Clustering");
+  tout.Branch("cluster", "std::vector<cluster>", &f_clust);
+  t->SetBranchAddress("dg_cell", &cell);
+  
+  for (int i = 0; i < nEvents; i++) {
+      //std::cout <<"EVENT: " << i <<std::endl; 
+    t->GetEntry(i);
+    std::vector<cluster> clust = Clusterize(std::move(cell));
+    
+    double CluEn = 0;
+    int n_clu = 0;
+    std::cout << "ENTRY: "<< i << ", FINAL CONFIGURATION: " << std::endl;
+    for (auto const& clu_info : clust) {
+        std::cout << "Cluster number: " << n_clu << std::endl;
+        Clust_info(clu_info);
+        std::cout << std::endl;
+        n_clu++;
+    }
```

Comment:
I'd remove line 26 to 32: it seems to me they are just a print-out for debugging.

Reply:
[5946f61](https://github.com/DUNE/sandreco/pull/27/commits/5946f61c2caca743e6c4b5719b7438fa6c7eaf52) 
commit solves this